packages/terra-pills/src/Pill.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNamesBind from 'classnames/bind';
+import {
+  KEY_SPACE,
+  KEY_RETURN,
+  KEY_DELETE,
+  KEY_BACK_SPACE,
+} from 'keycode-js';
+import ThemeContext from 'terra-theme-context';
+import IconClear from 'terra-icon/lib/icon/IconClear';
+import Popup from 'terra-popup';
+import styles from './Pill.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The label text of the pill component.
+   */
+  labelText: PropTypes.string.isRequired,
+  /**
+   * A callback function to execute when the pill is removed. If specified, the pill will be removable.
+   */
+  onRemove: PropTypes.func,
+  /**
+   * Object containing popup’s Content and the customizable attributes: contentHeight, contentWidth and isContentFocusDisabled.
+   */
+  popupConfig: PropTypes.shape({
+    /**
+     * The content to be displayed within the popup.
+     */
+    content: PropTypes.node.isRequired,
+    /**
+     * A string representation of the height in px, limited to:
+     * 40, 80, 120, 160, 240, 320, 400, 480, 560, 640, 720, 800, 880 or auto.
+     */
+    contentHeight: PropTypes.oneOf(Object.keys(Popup.Opts.heights)),
+    /**
+     * A string representation of the width in px, limited to:
+     * 160, 240, 320, 640, 960, 1280, 1760 or auto.
+     */
+    contentWidth: PropTypes.oneOf(Object.keys(Popup.Opts.widths)),
+    /**
+     * Set this to `true` if your content has focusable elements and you want them to receive focus instead of focusing on the default popup frame when the popup is opened.
+     */
+    isContentFocusDisabled: PropTypes.bool,
+  }),
+};
+
+const defaultProps = {
+  onRemove: undefined,
+  popupConfig: undefined,
+};
+
+const Pill = (props) => {
+  const {
+    labelText,
+    onRemove,
+    popupConfig,
+  } = props;
+
+  const [open, setOpen] = useState(false);
+  const pillRef = useRef();
+
+  const handleOnClick = () => {
+    setOpen(true);
+  };
+
+  const handleOnRemove = () => {
+    pillRef.current.style.display = 'none';
```

Comment:
This doesn't feel like a very "react" way to handle the remove functionality. Could we update this style based on an `isRemoved` state?

Reply:
I'd go a step farther and say that there's no need to hide the pill until the consumer stops rendering it, which is what they should be doing in their onRemove handler.

Reply:
This was just temporary for testing purposes for the 1st Iteration.
Have removed it now : https://github.com/cerner/terra-framework/pull/1260/commits/65bc8c8eb773f94746a47e8e3f29679e74fef500 and also updated test page to remove them from there.